src/python_bluesky_taskgraph/core/task.py

```diff
@@ -147,3 +150,29 @@ def _overwrite_results(self, results: List[Any] = None) -> None:
 
     def get_results(self, keys: List[str]) -> Dict[str, Any]:
         return {k: v for (k, v) in zip(keys, self._results) if k is not None}
+
+
+def run_stage_decorator(
+    func: Callable[[InputType], TaskOutput]
+) -> Callable[[InputType], TaskOutput]:
+    def decorated_func(args: InputType):
+        devices = {device for device in astuple(args) if isinstance(device, Device)}
```

Comment:
We don't want to tie this to the concrete `Device` as this will break when we use a non-ophyd library (such as ophyd v2). 
Suggest using the `bluesky.protocols.Stageable` interface instead.

```suggestion
        devices = {device for device in astuple(args) if isinstance(device, Stageable)}
```

Reply:
Done

Reply:
Idea for future refactor would be to remove ophyd references entirely from `core` and just use the interfaces but that's out-of-scope.